Index: lib/maruku/string_utils.rb
===================================================================
--- lib/maruku/string_utils.rb	(revision 160)
+++ lib/maruku/string_utils.rb	(working copy)
@@ -53,7 +53,7 @@
 			keys[:data] = $'
 			headers = $1
 			headers.split("\n").each do |l| 
-				k, v = l.split(':')
+				k, v = l.split(':', 2)
 				k, v = normalize_key_and_value(k, v)
 				k = k.to_sym
 #				puts "K = #{k}, V=#{v}"
